eo-maven-plugin/src/test/java/org/eolang/maven/AssembleMojoTest.java

```diff
@@ -215,32 +215,13 @@ void assemblesTogetherWithVersions(@TempDir final Path temp) throws Exception {
     }
 
     @Test
-    void assemblesNotFailWithFailOnErrorFlag(@TempDir final Path temp) throws IOException {
-        final Map<String, Path> result = new FakeMaven(temp)
-            .withProgram(AssembleMojoTest.INVALID_PROGRAM)
-            .with("failOnError", false)
-            .execute(AssembleMojo.class).result();
-        MatcherAssert.assertThat(
-            "Even if the eo program invalid we still have to parse it, but we didn't",
-            result.get(String.format("target/%s", ParseMojo.DIR)),
-            new ContainsFiles(String.format("**/main.%s", TranspileMojo.EXT))
-        );
-        MatcherAssert.assertThat(
-            "Since the eo program invalid we shouldn't have optimized it, but we did",
-            result.get(String.format("target/%s", OptimizeMojo.DIR)),
-            Matchers.not(new ContainsFiles(String.format("**/main.%s", TranspileMojo.EXT)))
-        );
-    }
-
-    @Test
-    void doesNotAssembleIfFailOnErrorFlagIsTrue(@TempDir final Path temp) {
-        final Class<IllegalStateException> expected = IllegalStateException.class;
+    void failsOnInvalidProgram(@TempDir final Path temp) {
         Assertions.assertThrows(
-            expected,
+            IllegalStateException.class,
             () -> new FakeMaven(temp)
                 .withProgram(AssembleMojoTest.INVALID_PROGRAM)
-                .execute(AssembleMojo.class),
-            String.format("AssembleMojo should have failed with %s, but didn't", expected)
+                .execute(new FakeMaven.Verify()),
```

Comment:
I think this whole test does not belong to the class. Here we should test `AssembleMojo`, not `VerifyMojo`. Let's move the test to the `VerifyMojoTest`. But here I believe we should add the test (if it does not exist yet) that would check that program with errors DOES NOT fail on assemble step

Reply:
@maxonfjvipon done